tests/cloudpickle_test.py

```diff
@@ -1078,129 +1080,167 @@ def f1():
-                cloudpickle.dump(mod.func_defined_in_dynamic_module, f)
-
-            child_process_code = """
-                import pickle
-
-                with open('function_with_initial_globals.pkl','rb') as f:
-                    func_with_initial_globals = pickle.load(f)
-
-                # At this point, a module called 'mod' should exist in
-                # _dynamic_modules_globals. Further function loading
-                # will use the globals living in mod.
-
-                assert func_with_initial_globals() == 'initial value'
-
-                # Load a function with initial global variable that was
-                # pickled after a change in the global variable
-                with open('function_with_modified_globals.pkl','rb') as f:
-                    func_with_modified_globals = pickle.load(f)
-
-                # assert the this unpickling did not modify the value of
-                # the local
-                assert func_with_modified_globals() == 'initial value'
-
-                # Update the value from the child process and check that
-                # unpickling again does not reset our change.
-                assert func_with_initial_globals('new value') == 'new value'
-                assert func_with_modified_globals() == 'new value'
-
-                with open('function_with_initial_globals.pkl','rb') as f:
-                    func_with_initial_globals = pickle.load(f)
-                assert func_with_initial_globals() == 'new value'
-                assert func_with_modified_globals() == 'new value'
-            """
-            assert_run_python_script(textwrap.dedent(child_process_code))
+        for override_existing_globals in [True, False]:
+            try:
+                # first, we create a dynamic module in the parent process
+                mod = types.ModuleType('mod')
+                code = '''
+                GLOBAL_STATE = "initial value"
+
+                def func_defined_in_dynamic_module(v=None):
+                    global GLOBAL_STATE
+                    if v is not None:
+                        GLOBAL_STATE = v
+                    return GLOBAL_STATE
+                '''
+                exec(textwrap.dedent(code), mod.__dict__)
+                # Simple sanity check on the function's output
+                assert mod.func_defined_in_dynamic_module() == "initial value"
+
+                # The function of mod is pickled two times, with two different
+                # values for the global variable GLOBAL_STATE.  Then we launch
+                # a child process that sequentially unpickles the two
+                # functions. Those unpickle functions should share the same
+                # global variables in the child process: Once the first
+                # function gets unpickled, mod is created and tracked in the
+                # child environment. This is state is preserved when unpickling
```

Comment:
... child environment. If override_globals is False, then the module state is preserved when unpickling...